src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -677,8 +699,155 @@ namespace OpenMS
       return EXECUTION_OK;
     }
 
-protected:
-    struct PeptideProteinMatchInformation
+ protected:
+
+     using MapDecoyStringToPrefixSuffixCount = std::map<std::string, std::pair<int, int>>;
+     using MapCaseInsensitiveToCaseSensitiveDecoyString = std::map<std::string, std::string>;
+
+     template<typename T>
+     ExitCodes findDecoyString_(FASTAContainer<T> & proteins,
+                                std::string & decoy_string,
+                                bool & is_prefix)
+     {
+       // Common decoy strings in fasta files. Used to determine if a protein is target/decoy automatically.
+       // Note: decoy prefixes/suffices must be provided in lower case and must only contain alphanumeric characters
+        MapDecoyStringToPrefixSuffixCount decoy_count = {{"decoy", {0,0}}, {"rev", {0,0}},
+                                                          {"reverse", {0,0}}, {"dev", {0,0}},
+                                                          {"iddecoy", {0,0}}, {"xxx", {0,0}},
+                                                          {"shuffle", {0,0}}, {"shuffled", {0,0}},
+                                                          {"random", {0,0}}};
+
+
+       // map case insensitive strings back to original case (as used in fasta)
+       MapCaseInsensitiveToCaseSensitiveDecoyString decoy_case_sensitive;
+
+       const size_t PROTEIN_CACHE_SIZE = 4e5;
+
+       while (true)
+       {
+         proteins.cacheChunk(PROTEIN_CACHE_SIZE);
+         bool has_active_data = proteins.activateCache();
```

Comment:
Just a comment: You use these on-time-variables quite often. Makes the code really hard to read IMHO
```
if (!proteins.activateCache())
{
  break; // easier?
}
```

Reply:
Yup, thanks. Noted.